src/fsharp/FSharp.Core/string.fs

```diff
@@ -31,12 +31,15 @@ namespace Microsoft.FSharp.Core
 
         [<CompiledName("Map")>]
         let map (mapping: char -> char) (str:string) =
-            if String.IsNullOrEmpty str then
-                String.Empty
+            if String.IsNullOrEmpty str then String.Empty
             else
-                let res = StringBuilder str.Length
-                str |> iter (fun c -> res.Append(mapping c) |> ignore)
-                res.ToString()
+                let result = str.ToCharArray()
+                let mutable i = 0
+                for c in result do
+                    result.[i] <- mapping c
+                    i <- i + 1
+
+                String result
```

Comment:
Please use explicit tostring()

Reply:
@forki, tx, but that would've output `"System.Char[]"`, but I've settled in https://github.com/dotnet/fsharp/pull/9470 with `new String(result)` to make it abundantly clear that the constructor is needed here.

In this PR it is not relevant anymore.